src/apis/controllers/products.controller.js

```diff
@@ -0,0 +1,201 @@
+          }, ${image}, ${price ?? 0}, ${quantity ?? 0}, ${ProductStatus.INSTOCK})
+        RETURNING id, name, slug
+      `;
+
+      return successResponse(
+        { newProduct },
+        "Create new product successfully",
+        200
+      );
+    }
+  ),
+
+  getAllProducts: controllerWrapper(
+    async (_, res, { errorResponse, successResponse, sql }) => {
+      const products = await sql`
+        SELECT id, category_id, name, slug, description, image, price, quantity, status
+        FROM products
+        WHERE deleted_at IS NULL
+      `;
+
+      return successResponse(
+        { products },
+        "Get all products successfully",
+        200
+      );
+    }
+  ),
+
+  // upload product
+  updateProduct: controllerWrapper(
+    async (req, _, { errorResponse, successResponse, sql }) => {
+      const { category_id, name, description, image, price, quantity, status } = req.body;
+      const { id } = req.params;
+
+      // check if product with name existed
+      const [existingProduct] = await sql`
+        SELECT id FROM products WHERE id = ${id} AND deleted_at IS NULL
+      `;
+
+      if(!existingProduct){
+        return errorResponse("Product does not exist", 404);
+      }
+
+      // check if product with name existed
+      // const [existingName] = await sql`
+      //   SELECT name, slug FROM products WHERE id = ${id}
+      // `;
+
+      // if (!existingName) {
+      //   return errorResponse("Product does not exist", 400);
+      // }
+
+      const [existingCategory] = await sql`
+        SELECT id FROM categories WHERE id = ${category_id}
+      `;
+
+      if (!existingCategory) {
+        return errorResponse(`Category with id ${category_id} not found`, 404);
+      }
+
+      const [product] = await sql`
+        UPDATE products
+        SET category_id = ${category_id} , name = ${name}, description = ${
+          description ?? ""
+        }, image = ${image}, price = ${price}, quantity = ${quantity}, status = ${status}
+        WHERE id = ${id}
+        RETURNING id, category_id, name, slug, description, image, price, quantity, status
+      `;
+
+      if (!product) {
+        return errorResponse(`Product with id ${id} not found`, 404);
+      }
+
+      return successResponse(
+        { product },
+        "Upload product image successfully",
+        200
+      );
+    }
+  ),
+
+  // SetDelete Product
+  deleteProduct: controllerWrapper(
+    async (req, _, { errorResponse, successResponse, sql }) => {
+      const { id } = req.params;
+
+      const [existingProduct] = await sql`
+        SELECT id FROM products WHERE id = ${id}
+      `;
+
+      if (!existingProduct) {
+        return errorResponse(`Product with id ${id} not found`, 404);
+      }
+
+      const [isDelted] = await sql`
+        SELECT id FROM products WHERE id = ${id} AND deleted_at IS NULL
+      `;
+
+      if (!isDelted) {
+        return errorResponse(`Product with id ${id} is deleted`, 404);
+      }
```

Comment:
Query 1 time -> `SELECT id, deleted_at` check not found and `deleted_at` for is deleted